docker/checkout.bash

```diff
@@ -69,16 +71,38 @@ echo "" > COLCON_IGNORE
 
 cd ../
 
+# Clone the foxy branch of ros2_tracing in order to enable certain analyses of CARMA Platform
+# made possible through collected trace data, such as analyzing ROS 2 callback durations.
+git clone -b foxy https://github.com/ros2/ros2_tracing
+
 #rosbridge_suite is a ROS meta-package including all the rosbridge packages.
 # NOTE: clone -b flag is used instead of --branch to avoid hook rewriting it
 git clone -b ros2 https://github.com/usdot-fhwa-stol/rosbridge_suite
 
+# The feature/integrate-carma branch of rosbag2 includes improvements that were not possible to backport into the foxy branch
+# of rosbag2. These rosbag2 packages will replace the originally built foxy rosbag2 packages.
+# NOTE: Additional information regarding the rosbag2 improvements on this branch are included in the forked repository's README.
+git clone -b carma-develop https://github.com/usdot-fhwa-stol/rosbag2
```

Comment:
we are targeting `carma-develop` erroneously here.
need `carma-system-4.5.0` anticipating we will create such tag on rosbag2 forked repo